t/access_ht_total_user_switched.t

```diff
@@ -0,0 +1,238 @@
+
+use Test::File;
+
+use Data::Dumper;
+use feature qw(say);
+
+#---- MONEKYPATCHES
+no warnings 'redefine';
+
+{
+
+    package Session;
+
+    sub new {
+        my $class = shift;
+        return bless {}, $class;
+    }
+
+    sub get_persistent {
+        my ( $self, $key ) = @_;
+        return $$self{$key};
+    }
+
+    sub set_persistent {
+        my ( $self, $key, $value ) = @_;
+        $$self{$key} = $value;
+    }
+
+}
+
+
+local *Auth::Auth::affiliation_is_hathitrust = sub {
+    return 1;
+};
+
+local *Auth::Auth::auth_sys_is_SHIBBOLETH = sub {
+    return 1;
+};
+
+local *Auth::ACL::S___total_access_using_DEBUG_super = sub {
+    return 0;
+};
+
+local *Auth::ACL::a_Authorized = sub {
+    my $access_ref = shift;
+    return ( defined $$access_ref{access} && $$access_ref{access} eq 'total' );
+};
+
+local *Auth::Auth::affiliation_has_emergency_access = sub {
+return 0;
+};
+
+#---- MONEKYPATCHES
+
+my $C   = new Context;
+my $cgi = new CGI;
+$C->set_object( 'CGI', $cgi );
+my $config = new MdpConfig(
+    File::Spec->catdir( $ENV{SDRROOT}, 'mdp-lib/Config/uber.conf' ),
+    File::Spec->catdir( $ENV{SDRROOT}, 'slip-lib/Config/common.conf' )
+);
+$C->set_object( 'MdpConfig', $config );
+
+my $db_user = $ENV{'MARIADB_USER'} || 'ht_testing';
+my $db      = new Database($db_user);
+$C->set_object( 'Database', $db );
+
+my $auth = Auth::Auth->new($C);
+$C->set_object( 'Auth', $auth );
+
+my $sess = Session->new($C);
+$C->set_object('Session', $sess);
+
+mock_institutions($C);
+mock_acls($C);
+
+local %ENV = %ENV;
+$ENV{HTTP_HOST}   = q{babel.hathitrust.org};
+$ENV{SERVER_ADDR} = q{141.213.128.185};
+$ENV{SERVER_PORT} = q{443};
+$ENV{AUTH_TYPE}   = q{shibboleth};
+$ENV{affiliation} = q{member@umich.edu};
+
+sub setup_us_institution {
+    $ENV{REMOTE_USER}            = 'user@umich.edu';
+    $ENV{eppn}                   = q{user@umich.edu};
+    $ENV{umichCosignFactor}      = q{UMICH.EDU};
+    $ENV{Shib_Identity_Provider} = Auth::Auth::get_umich_IdP_entity_id();
+}
+
+sub setup_nonus_instition {
+    $ENV{REMOTE_USER} = 'user@ox.ac.edu';
+    $ENV{eppn}        = q{user@ox.ac.edu};
+    delete $ENV{umichCosignFactor};
+    $ENV{Shib_Identity_Provider} = q{https://registry.shibboleth.ox.ac.uk/idp};
+    $ENV{affiliation}            = q{member@ox.ac.edu};
+    $ENV{entitlement} = q{http://www.hathitrust.org/access/enhancedText};
+}
+
+sub test_attr {
+    my ( $attr, $access_profile, $location ) = @_;
+    my $id = "test.$attr\_$access_profile";
+    $ENV{TEST_GEO_IP_COUNTRY_CODE} = $location || 'US';
+
+    unless ($attr) {
+        print STDERR caller();
+    }
+
+    my $ar     = Access::Rights->new( $C, $id );
+    my $status = $ar->check_final_access_status( $C, $id );
+    return $status;
+}
+
+my $num_tests = 0;
+
+my $tests =
+  Test::File::load_data("$FindBin::Bin/data/access/ht_total_user.tsv");
+
+my $ht_affiliate_tests =
+  Test::File::load_data("$FindBin::Bin/data/access/ht_affiliate.tsv");
+
+my $ht_affiliate_expected = {};
```

Comment:
I'm not sure need to re-test everything here -- since (I think) we already test elsewhere that `HT_TOTAL_USER` vs. `HT_AFFILIATE` get the behavior we expect, we really only need to test here that the given environment & session get `HT_AFFILIATE` without the toggle and `HT_TOTAL_USER` with the toggle. (That said, if we had end-to-end integration/feature tests, I think a useful one for this feature would be setting up the environment for this user, verifying they can't see IC material but can see the toggle, then clicking the toggle & verifying they can see the IC material, then un-toggling it and verifying they can't any more..)

Reply:
The current `HT_TOTAL_USER` test assumes a binary state --- you've enabled `debug=super` --- and this inches toward testing the toggle as well as access (you _could_ have access but currently don't). And the data files only give us the `allow` value, not which attributes require using the switch.

Reply:
:+1: